### PR TITLE
chore: RDF module with types

### PR DESCRIPTION
**Pull request by victordelval**
## Description

Closes: regen-network/regen-registry#1367

(*) To better understand the context of this proposal, take a look at issue [#1357](regen-network/regen-registry#1357) , which illustrates a complete proposal at the web-registry level (mainly `lib`, but also `core`)

The files related to **rdf** and **metadata/json-ld** types/query have been reordered, making a clear distinction between the rdf library (as utility, business agnostic) and the json-ld types corresponding to the metadata stored in the offchain DB, with its corresponding api query

* `lib/rdf` module. With the rdf files, as well as the generic types and also the types defined for the `rdf-validate-shacl` dependency.
* `lib/db` module (off-chain database). For now it only contains things related to metadata:
  * `api` with metadata-graph query
  * `types/json-ld` with generated metadata types, and unions/intersections

**TODO**  ~A change is still pending. Due to the migration of the files related to rdf located in web-components, now in web-registry, there is a small part of code that is used by 3 UI components located in web-components (`RoleField`, `CertificationForm` and `ProfileModal`), so has kept. We just need to relocate these components in web-registry.~
Right now only `ProfileModal` has been kept in web-components, hiding a duplicate interface `UrlType`, since it is a component dependency of several components there.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1612--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by victordelval** [comment]
inside `web-registry/lib`

rdf_module
db_module

File: web-components/src/utils/schemaURL.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
+// TODO: type UrlType and getURLInitialValue
+// this has been temporarily duplicated due to moving the RDF types to the web-registry app
+
 export interface UrlType {
   '@type': 'schema:URL';
   '@value'?: string | null;
 }
```
Comment (blushi) [inline web-components/src/utils/schemaURL.ts:7]: I believe we could even just merge `UrlType` and `URL` into one:

```
export interface UrlType {
  '@type': 'http://schema.org/URL' | 'schema:URL';
  '@value'?: string | null;
}
```

**Review by mhagel — APPROVED** [review]
I like this organization. Thanks!

**Comment by victordelval** [comment]
@Mergifyio backport redwood hambach

File: web-components/src/utils/schemaURL.ts (hunk 1 above)
Comment (victordelval) [reply web-components/src/utils/schemaURL.ts]: In the end I preferred not to refactor any of the code itself. I have it on the radar for future refactorings..

**Pull request merged**